### PR TITLE
Display cookie message to EU browsers

### PR DESCRIPTION
**Pull request by JustinPinner**
We have to display a message about the use of cookies to EU readers.

We'll show it on the first page view, if the user hasn't seen the message before and if they've arrived from an EU location.

There's no close button on the message. It will disappear when they browse to a new page.

There is a corresponding PR in Fastly (https://github.com/guardian/fastly-edge-cache/pull/490) to provide the continent code.

Here are a couple of example screen grabs;
screen shot 2015-10-14 at 11 28 52

screen shot 2015-10-14 at 11 30 14

### CODE REVIEW

**Comment by ScottPainterGNM** [comment]
Do you think maybe not having a dismiss button might cause users to complain before they realise it will disappear automatically? Is having a dismiss button in line with the cookie notice rules?

**Comment by davidrapson** [comment]
:scream_cat: "We have to display a message about the use of cookies to EU readers" Is this really still true?

**Comment by JustinPinner** [comment]
@ScottPainterGNM I know. I initially built it with the close button visible but our PM said that it mustn't be any kind of call to action. So it went.

**Comment by JustinPinner** [comment]
@davidrapson I can find lots of places that point to explanations of the law, and none (that I've read so far) say it has been repealed. It seems that it has been a requirement since 2012. Here's how the ICO do it currently;
screen shot 2015-10-14 at 12 25 51

**Comment by paperboyo** [comment]
> Do you think maybe not having a dismiss button might cause users to complain before they realise it will disappear automatically?

I do. I will :wink:

Also, "Welcome" clashes with the batman logo on desktop BP in the screenshots above.

**Comment by Jholder112233** [comment]
This definitely needs a dismiss button.

**Comment by JustinPinner** [comment]
> This definitely needs a dismiss button

@stephanfowler - reinstate the close button? Also, drop the brand logo? Both of these sound sensible to me.

The only design guideline I've seen is from the R2 equivalent;
screen shot 2015-10-14 at 13 46 54

**Comment by JustinPinner** [comment]
Updated screen grabs;

Mobile:
screen shot 2015-10-15 at 10 26 41

Tablet:
screen shot 2015-10-15 at 10 27 48

Desktop:
screen shot 2015-10-15 at 10 28 25

**Comment by paperboyo** [comment]
Terribly sorry to be picky, but the X seems not vertically centred in the close button...

**Comment by NataliaLKB** [comment]
Once the x is centred, :+1: . It looks much better!

**Comment by Jholder112233** [comment]
This is actually a really big change - going to be seen by every new user. Just checking this has this gone through @zeftilldeath or @Pearson82? Seems like the left padding is misaligned.

I also thought we were going to get the copywriter involved to make the text 'playfully intelligent'?

File: static/src/javascripts/projects/common/modules/ui/cookiesBanner.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            impressions = cookieVal && !isNaN(cookieVal) ? parseInt(cookieVal, 10) : 0;
+            return impressions == 0;
+        } else {
+            return false;
+        }
+    }
+
+    function showMessage() {
+        var link = 'https://www.theguardian.com/info/cookies',
+            txt = 'Welcome to the Guardian. This site uses cookies, read our policy <a href="' + link + '" class="cookie-message__link">here</a>.',
```
Comment (Jholder112233) [inline static/src/javascripts/projects/common/modules/ui/cookiesBanner.js]: I do believe Guardian style would be to not have a full stop on the end here

**Comment by JustinPinner** [comment]
Ok, if someone can give me a hand to tweak the CSS according to our usual methods I'd appreciate it because I haven't altered anything (knowingly) that alters the appearance of the close button's X character, so what we're seeing is, I think, the out-of-the-box behaviour. @Jholder112233 The copy came from Piers, and I don't think any design eyes have been involved.

File: static/src/javascripts/projects/common/modules/ui/cookiesBanner.js (hunk 1 above)
Comment (JustinPinner) [inline static/src/javascripts/projects/common/modules/ui/cookiesBanner.js]: This text is exactly as provided by Piers. Fullstop included.

**Comment by Pearson82** [comment]
I did supply a design for this at some stage but this should just be using our standard messaging and button styles so seems odd that things aren't aligned. @zeftilldeath do you think you'd be able to finesse this a bit?

**Comment by JustinPinner** [comment]
We're going to have to try an alternative method of obtaining the continent code from Fastly (see the note on the Fastly PR). The code here is reasonable and may not have to change a great deal, but rather than leave this PR hanging around I'll close it until plan B is ready.

**Pull request closed without merging**